src/Bicep.Types/Concrete/FunctionType.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+
+namespace Azure.Bicep.Types.Concrete;
+
+public class FunctionType : TypeBase
+{
+    [JsonConstructor]
+    public FunctionType(string name, ITypeReference output, IReadOnlyList<FunctionTypeParameter> parameters, string? description)
```

Comment:
Is the name here the function name? or a name for the type?

Reply:
We don't really have a strong concept of naming types in Bicep - so the former. Good point about separating the two out though - it is confusing to have this be a property of the types.